### PR TITLE
28 documentation for pbparam

### PR DESCRIPTION
**Pull request by muhammedsogut**
# Description

This PR is opened for documentation on how to use pbparam. It provides docs files and explanations on the functions used. The documents are created using SphinX. Also, example notebooks are redesigned to work on Google Colab.

Fixes # (issue)

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/paramm-team/pybamm-param/blob/main/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -10,9 +10,10 @@
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 #
-# import os
-# import sys
-# sys.path.insert(0, os.path.abspath('.'))
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath("../"))
```
Comment (brosaplanella) [inline docs/conf.py:16]: It seems we can leave this commented and the docs still work. Can you double check this and, if so, comment these lines back again?

File: docs/index.rst [hunk 2]
```diff
@@ -6,20 +9,76 @@
 Welcome to pybamm-param's documentation!
 ========================================
 
+Python Battery Mathematical Modelling (**PyBAMM**) Parameter Optimisation package is developed for optimising battery parameters of
```
Comment (brosaplanella) [inline docs/index.rst:12]:
```suggestion
Python Battery Mathematical Modelling (**PyBaMM**) Parameter Optimisation package is developed for optimising battery parameters of
```

File: docs/index.rst [hunk 3]
```diff
@@ -6,20 +9,76 @@
...
+PyBAMM model parameters using various optimisation methods and error functions.
```
Comment (brosaplanella) [inline docs/index.rst:13]:
```suggestion
PyBaMM model parameters using various optimisation methods and error functions. 
```

File: docs/index.rst [hunk 4]
```diff
@@ -6,20 +9,76 @@
 Welcome to pybamm-param's documentation!
 ========================================
 
+Python Battery Mathematical Modelling (**PyBAMM**) Parameter Optimisation package is developed for optimising battery parameters of
+PyBAMM model parameters using various optimisation methods and error functions. 
+
+PyBAMM-param is hosted on Github_. This page provides the *API*, or *developer documentation* for ''pybamm-param''.
```
Comment (brosaplanella) [inline docs/index.rst:15]:
```suggestion
pybamm-param is hosted on Github_. This page provides the *API*, or *developer documentation* for ''pybamm-param''.
```

File: docs/index.rst [hunk 5]
```diff
@@ -6,20 +9,76 @@
...
+
+Detailed examples can be viewed on the
+`GitHub examples page <https://github.com/paramm-team/pybamm-param/tree/main/examples/notebooks>`_,
+and run locally using ``jupyter notebook``, or online through
+`Google Colab <https://colab.research.google.com/github/paramm-team/>`_.
+
+Contributing
+============
+
+Contributions to PyBaMM and its development are welcome! If you have ideas for features, bug fixes, models, spatial methods, or solvers, we would love to hear from you.
```
Comment (brosaplanella) [inline docs/index.rst:82]:
```suggestion
Contributions to pybamm-param and its development are welcome! If you have ideas for features, bug fixes, models, spatial methods, or solvers, we would love to hear from you.
```

**Review by brosaplanella — APPROVED** [review]
Looks good, just a few comments and suggestions!

File: docs/conf.py (hunk 1 above)
Comment (muhammedsogut) [reply docs/conf.py:16]: SphinX couldn't find the pbparam files when comment them out. Therefore, I'm leaving as it is as agreed.

**Pull request merged**